src/Browscap/Data/Factory/DataCollectionFactory.php

```diff
@@ -0,0 +1,272 @@
+<?php
+declare(strict_types = 1);
+namespace Browscap\Data\Factory;
+
+use Assert\Assertion;
+use Browscap\Data\DataCollection;
+use Browscap\Data\Validator\DivisionDataValidator;
+use Psr\Log\LoggerInterface;
+
+class DataCollectionFactory
```

Comment:
factories can be declared as `final`

Reply:
This factory is mocked to keep the tests small. So we can't `final` it.